ncm-ipmi/src/main/pan/components/ipmi/schema.pan

```diff
@@ -2,25 +2,74 @@
 # ${developer-info}
 # ${author-info}
 
-
+@{
+    See Intelligent Platform Management Interface Specification v2.0 rev. 1.1
+    https://www.intel.co.uk/content/www/uk/en/products/docs/servers/ipmi/ipmi-second-gen-interface-spec-v2-rev1-1.html
+}
 declaration template components/ipmi/schema;
 
 include 'quattor/schema';
 
 type structure_users = {
-    "login" : string
-    "password" : string
-    "priv" ? string
-    "userid" ? long
+    @{
+        User Name String in ASCII (16 bytes)
+        See IPMI Spec v2.0 rev 1.1 section 22.28 Set User Name Command
+    }
+    "login" : string_trimmed(1..16) with match(SELF, '^[ -~]+$')
```

Comment:
what is this match supposed to do?

Reply:
Restrict the field to only printable ASCII characters, which happens to be from space to to tilde.

Reply:
It could instead be `^[\x20-\x7e]+$` if preferred.

Reply:
The hex-code does make it more obvious it's a range, if less nice to read.

Reply:
AFAICT POSIX character classes aren't supported, but if they are then `^[[:print:]]+$`.

Reply:
Ah, of course there's a special Java-way of doing POSIX character classes: `^\p{Print}+$`.

Reply:
i thought as much, it should have been `match(SELF, '^[^ -~]+$')` :wink:

Reply:
ah, so it really was intended as a range. :shivers: